File: day-02/PartOne.cs
// SPDX-FileCopyrightText: 2020 Pablo Jiménez Pascual <pablo@jimpas.me>
//
// SPDX-License-Identifier: GPL-3.0-or-later

using System;
using System.Linq;

namespace day_02
{
    class PartOne
    {
        static void Main(string[] args)
        {
            char[] delimiters = { '-', ' ', ':' };
            int cont = 0;
            string[] lines = System.IO.File.ReadAllLines(args[0]);
            foreach (string line in lines)
            {
                string[] data = line.Split(delimiters);
                int min = Int32.Parse(data[0]);
                int max = Int32.Parse(data[1]);
                char target = char.Parse(data[2]);
                string text = data[4];
                int freq = text.Count(f => (f == target));
                if (freq >= min && freq <= max)
                {
                    cont++;
                }
            }
            Console.WriteLine(cont);
        }
    }
}

File: day-05/part_one.py
# SPDX-FileCopyrightText: 2020 Pablo Jiménez Pascual <pablo@jimpas.me>
#
# SPDX-License-Identifier: GPL-3.0-or-later

import sys

def convert_to_dec(text):
    dec = 0
    for c in text:
        dec = dec * 2 + 1
        if c == 'F' or c == 'L':
            dec = dec - 1
    return dec


def seat_id(line):
    row = convert_to_dec(line[:7])
    column = convert_to_dec(line[7:10])
    return row * 8 + column


with open(sys.argv[1], "r") as file:
    max = 0
    for line in file:
        current = seat_id(line)
        if current > max:
            max = current
    print(max)
